packages/core/lib/command-utils.js

```diff
@@ -206,9 +212,85 @@ const displayGeneralHelp = () => {
     .showHelp();
 };
 
+const getConfiguredNetworkUrl = function (customConfig, isDashboardNetwork) {
+  const defaultPort = isDashboardNetwork
+    ? managedDashboardDefaultPort
+    : managedGanacheDefaultPort;
+  const configuredNetworkOptions = {
+    host: customConfig.host || defaultHost,
+    port: customConfig.port || defaultPort
+  };
+  const urlSuffix = isDashboardNetwork ? "/rpc" : "";
+  return `http://${configuredNetworkOptions.host}:${configuredNetworkOptions.port}${urlSuffix}`;
+};
+
+const deriveConfigEnvironment = function (detectedConfig, network, url) {
+  let configuredNetwork;
+
+  const configDefinesProvider =
+    detectedConfig.networks[network] &&
+    detectedConfig.networks[network].provider;
+
+  if (configDefinesProvider) {
+    // Use "provider" specified in the config to connect to the network
+    // along with the other specified network properties
+    configuredNetwork = {
+      ...detectedConfig.networks[network],
+      network_id: "*",
+      provider: detectedConfig.networks[network].provider
+    };
+  } else if (url) {
+    // Use "url" to configure network (implies not "develop" and not "dashboard")
+    configuredNetwork = {
+      network_id: "*",
+      url,
+      provider: function () {
+        return new Web3.providers.HttpProvider(url, {
+          keepAlive: false
+        });
+      }
+    };
+  } else {
+    // Otherwise derive network settings
+    const customConfig = detectedConfig.networks[network] || {};
+    const isDashboardNetwork = network === "dashboard";
+    const configuredNetworkUrl = getConfiguredNetworkUrl(
+      customConfig,
+      isDashboardNetwork
+    );
+    const defaultPort = isDashboardNetwork
+      ? managedDashboardDefaultPort
+      : managedGanacheDefaultPort;
+    const defaultNetworkId = isDashboardNetwork
+      ? "*"
+      : managedGanacheDefaultNetworkId;
+
+    configuredNetwork = {
+      // customConfig will spread only when it is defined and ignored when undefined
+      ...customConfig,
```

Comment:
Do we not want to spread after adding the defaults? That way user settings override the default ones.

Reply:
I think your suggestion makes sense @eggplantzzz. We can spread the `customConfig` after the defaults.